di-ipv-credential-issuer-stub/src/main/java/uk/gov/di/ipv/stub/cred/handlers/AuthorizeHandler.java

```diff
@@ -209,43 +212,37 @@ private void persistData(QueryParamsMap queryParamsMap, AuthorizationCode author
     }
 
     private ValidationResult validateQueryParams(QueryParamsMap queryParams) {
-        String redirectUriValue = queryParams.value(RequestParamConstants.REDIRECT_URI);
-        if (Validator.isNullBlankOrEmpty(redirectUriValue)) {
-            return new ValidationResult(false, new ErrorObject(
-                    ERROR_CODE_INVALID_REDIRECT_URI,
-                    "redirect_uri param must be provided",
-                    HttpServletResponse.SC_BAD_REQUEST));
-        }
-
         String responseTypeValue = queryParams.value(RequestParamConstants.RESPONSE_TYPE);
         if (Validator.isNullBlankOrEmpty(responseTypeValue) || !responseTypeValue.equals(ResponseType.Value.CODE.getValue())) {
             return new ValidationResult(false, OAuth2Error.UNSUPPORTED_RESPONSE_TYPE);
         }
 
         String clientIdValue = queryParams.value(RequestParamConstants.CLIENT_ID);
-        if (Validator.isNullBlankOrEmpty(clientIdValue)) {
+        if (Validator.isNullBlankOrEmpty(clientIdValue) || CredentialIssuerConfig.getClientConfig(clientIdValue) == null) {
             return new ValidationResult(false, OAuth2Error.INVALID_CLIENT);
         }
 
+        if (Validator.redirectUrlIsInvalid(queryParams)) {
+            return new ValidationResult(false, new ErrorObject(
+                    ERROR_CODE_INVALID_REDIRECT_URI,
+                    "redirect_uri param must be provided",
```

Comment:
If you agree with the above then this error description should maybe be changed slightly to be specifcally something like 

```suggestion
                    "redirect_uri param provided does not match any of the allowed redirect_uri values configured",
```

Reply:
Having looked at the error objects provided by nimbus, I think I'm going to change these to use a standard one. Maybe add some logging to make it easier to work out what went wrong.

Reply:
Actually, perhaps not...